rust/parser/typeql.pest

```diff
@@ -102,220 +81,329 @@ pattern_try = { TRY ~ CURLY_OPEN ~ patterns ~ CURLY_CLOSE }
-annotations_owns = { ( annotation_card | ANNOTATION_DISTINCT | ANNOTATION_KEY | ANNOTATION_UNIQUE )* }
-annotations_relates = { ( annotation_card | ANNOTATION_DISTINCT | ANNOTATION_CASCADE )* } // ( ANNOTATION_ABSTRACT )?
-annotations_value = { ( annotation_regex | annotation_values )* }
-
-annotation_card = { ANNOTATION_CARD ~ PAREN_OPEN ~ ANNOTATION_CARD_LOWER ~ COMMA ~ ANNOTATION_CARD_UPPER ~ PAREN_CLOSE }
-annotation_regex = { ANNOTATION_REGEX ~ PAREN_OPEN ~ QUOTED_STRING ~ PAREN_CLOSE }
-annotation_subkey = { ANNOTATION_SUBKEY ~ PAREN_OPEN ~ label ~ PAREN_CLOSE }
-annotation_values = { ANNOTATION_VALUES ~ PAREN_OPEN ~ value_primitive ~ ( COMMA ~ value_primitive )* ~ PAREN_CLOSE }
-
 // THING STATEMENTS ============================================================
 
-statement_things = { ( statement_thing ~ SEMICOLON )+ }
-statement_thing = { thing_start ~ ( COMMA ~ thing_constraint )* }
+statement_thing = { statement_thing_var | statement_anon_relation }
+statement_anon_relation = { relation ~ thing_constraint? ~ ( COMMA ~ thing_constraint )* }
 
-thing_start = { relation
-              | VAR ~ thing_constraint
-              | VAR ~ value_primitive ~ ISA_ ~ type_ref
-              | VAR ~ expression_struct ~ ISA_ ~ type_ref
-              | VAR ~ comparison ~ ISA_ ~ type_ref
-              }
+statement_thing_var = { var ~ thing_constraint ~ ( COMMA ~ thing_constraint )*
+                      | var ~ ( value_literal | expression_struct ) ~ isa_constraint
+                      | var ~ comparison ~ isa_constraint
+                      }
 
-thing_constraint = { ISA_ ~ type_ref
-                   | IID ~ IID_VALUE
-                   | relates_constraint
-                   | has_constraint
-                   }
+thing_constraint = { isa_constraint | iid_constraint | has_constraint | links_constraint }
 
-has_constraint = { HAS ~ ( VAR | label ~ ( expression_value | comparison ) )
-                 | HAS ~ type_ref_list ~ ( expression_list | comparison )
+isa_constraint = { ISA_ ~ type_ref }
+iid_constraint = { IID ~ iid_value }
+has_constraint = { HAS ~ type_ref_list ~ ( comparison | expression_list | var )
+                 | HAS ~ type_ref ~ ( comparison | expression_value | var )
+                 | HAS ~ var
                  }
-relation_constraint = { LINKS ~ relation }
+links_constraint = { LINKS ~ relation }
 
 relation = { PAREN_OPEN ~ role_player ~ ( COMMA ~ role_player )* ~ PAREN_CLOSE } // A list of role players in a Relations
-role_player = { type_ref_list ~ COLON ~ VAR
-              | type_ref ~ COLON ~ VAR
-              | VAR
+role_player = { type_ref_list ~ COLON ~ var
+              | type_ref ~ COLON ~ var
+              | var
               }
 
-// COMPARISON CONSTRUCTS =======================================================
+statement_deletable = { HAS ~ var ~ OF ~ var
+                      | LINKS ~ relation ~ OF ~ var
+                      | var
+                      }
```

Comment:
Ok we decided to make the `HAS` and `LINKS` optional, so users can just write `delete $a of $a;` and `delete (friend: $x) of $r;`